core/dataset.cc

```diff
@@ -551,8 +551,12 @@ queryStructured(const SelectExpression & select,
     std::mutex lock;
     std::vector<MatrixNamedRow> output;
 
+    if (!having.isConstantTrue() && groupBy.clauses.empty())
+        throw HttpReturnException(400, "HAVING expression requires a GROUP BY expression");
```

Comment:
Is that true?  What about something with an aggregator but no group by... is HAVING illegal there?